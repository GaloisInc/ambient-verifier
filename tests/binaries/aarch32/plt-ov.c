#include <unistd.h>

#include <sys/types.h>

/*
 * Test that the verifier can identify PLT stubs and dispatch overrides for
 * shared library functions.
 */

int main(void) {
  // On ARM the creation of a getpid function pointer has no effect on whether
  // a stub for getpid is placed in .plt.got.  It is still placed in .plt.
  pid_t (*g)() = &getpid;
  pid_t pid = getpid();
  pid_t ppid = getppid();

  return 0;
}
