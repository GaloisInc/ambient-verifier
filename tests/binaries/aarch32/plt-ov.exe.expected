successful: 1
failed: 0
overrideDir: tests/overrides/
